OpenRCT2 does not build on Arch Linux

**OS:** Arch Linux x64
**Version:** 0.2.6
**Commit/Build:** 8798811561b71c7792ac38cd6c14580c2315582b

When building OpenRCT2 on Arch Linux a compilation error occurs:

```
[ 99%] Building CXX object CMakeFiles/openrct2.dir/src/openrct2-ui/windows/ViewClipping.cpp.o
[100%] Building CXX object CMakeFiles/openrct2.dir/src/openrct2-ui/windows/Viewport.cpp.o
[100%] Building CXX object CMakeFiles/openrct2.dir/src/openrct2-ui/windows/Water.cpp.o
[100%] Linking CXX executable openrct2
D.21324]’/home/ccoors/.cache/yay/openrct2-git/src/openrct2-git/src/openrct2-ui/scripting/ScViewport.hpp: In member function ‘moveTo’:
/home/ccoors/.cache/yay/openrct2-git/src/openrct2-git/src/openrct2-ui/scripting/ScViewport.hpp:302:21: error:  may be used uninitialized in this function [-Werror=maybe-uninitialized]
  302 |             return {};
      |                     ^
D.21324 + 4B]’/home/ccoors/.cache/yay/openrct2-git/src/openrct2-git/src/openrct2-ui/scripting/ScViewport.hpp:302:21: error:  may be used uninitialized in this function [-Werror=maybe-uninitialized]
D.21324 + 8B]’/home/ccoors/.cache/yay/openrct2-git/src/openrct2-git/src/openrct2-ui/scripting/ScViewport.hpp:302:21: error:  may be used uninitialized in this function [-Werror=maybe-uninitialized]
lto1: all warnings being treated as errors
lto-wrapper: fatal error: /usr/bin/c++ returned 1 exit status
compilation terminated.
/usr/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/openrct2.dir/build.make:1864: openrct2] Error 1
make[1]: *** [CMakeFiles/Makefile2:251: CMakeFiles/openrct2.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
Error making: openrct2-git
```
The error message is confusing as it does not contain the variable name.

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. Try to build the AUR package `openrct2-git`/try to build from source using `cmake .. && make`.

--- a/src/openrct2-ui/scripting/CustomWindow.cpp
+++ b/src/openrct2-ui/scripting/CustomWindow.cpp
@@ -650,7 +650,7 @@ namespace OpenRCT2::Ui::Windows
             }
             widgetIndex++;
         }
-        return 0;
+        return std::nullopt;
     }
 
     static void window_custom_update_viewport(rct_window* w)
@@ -978,7 +978,7 @@ namespace OpenRCT2::Ui::Windows
                 }
             }
         }
-        return {};
+        return std::nullopt;
     }
 
 } // namespace OpenRCT2::Ui::Windows

--- a/src/openrct2-ui/scripting/ScViewport.hpp
+++ b/src/openrct2-ui/scripting/ScViewport.hpp
@@ -299,7 +299,7 @@ namespace OpenRCT2::Scripting
                     }
                 }
             }
-            return {};
+            return std::nullopt;
         }
     };
 } // namespace OpenRCT2::Scripting

--- a/src/openrct2/interface/Screenshot.cpp
+++ b/src/openrct2/interface/Screenshot.cpp
@@ -141,7 +141,7 @@ static std::optional<std::string> screenshot_get_next_path()
     if (!platform_ensure_directory_exists(screenshotDirectory.c_str()))
     {
         log_error("Unable to save screenshots in OpenRCT2 screenshot directory.");
-        return {};
+        return std::nullopt;
     }
 
     auto parkName = screenshot_get_park_name();
@@ -165,7 +165,7 @@ static std::optional<std::string> screenshot_get_next_path()
     }
 
     log_error("You have too many saved screenshots saved at exactly the same date and time.");
-    return {};
+    return std::nullopt;
 };
 
 std::string screenshot_dump_png(rct_drawpixelinfo* dpi)

--- a/src/openrct2/network/ServerList.cpp
+++ b/src/openrct2/network/ServerList.cpp
@@ -92,7 +92,7 @@ std::optional<ServerListEntry> ServerListEntry::FromJson(const json_t* server)
     if (name == nullptr || version == nullptr)
     {
         log_verbose("Cowardly refusing to add server without name or version specified.");
-        return {};
+        return std::nullopt;
     }
     else
     {

--- a/src/openrct2/peep/Peep.cpp
+++ b/src/openrct2/peep/Peep.cpp
@@ -610,7 +610,7 @@ std::optional<CoordsXY> Peep::UpdateAction(int16_t& xy_distance)
     {
         if (xy_distance <= destination_tolerance)
         {
-            return {};
+            return std::nullopt;
         }
         int32_t nextDirection = 0;
         if (x_delta < y_delta)

--- a/src/openrct2/rct2/S6Exporter.cpp
+++ b/src/openrct2/rct2/S6Exporter.cpp
@@ -1622,7 +1622,7 @@ std::optional<uint16_t> S6Exporter::AllocateUserString(const std::string_view& v
         _userStrings.emplace_back(value);
         return static_cast<uint16_t>(USER_STRING_START + nextId);
     }
-    return {};
+    return std::nullopt;
 }
 
 static std::string GetTruncatedRCT2String(const std::string_view& src)

--- a/src/openrct2/ride/Vehicle.cpp
+++ b/src/openrct2/ride/Vehicle.cpp
@@ -2210,7 +2210,7 @@ static std::optional<uint32_t> ride_get_train_index_from_vehicle(Ride* ride, uin
         }
         else if (trainIndex >= std::size(ride->vehicles))
         {
-            return {};
+            return std::nullopt;
         }
     }
     return { trainIndex };

